src/higher-level/index-hl.c

```diff
@@ -136,6 +136,107 @@ float tmp, ind, scale = 10000.0;
 }
 
 
+/** This function computes a spectral index time series, with a Ratio - 1,
++++ e.g. CIre: (b1/b2)-1
+--- ard:    ARD
+--- mask_:  mask image
+--- ts:     pointer to instantly useable TSA image arrays
+--- b1:     band 1
+--- b2:     band 2
+--- nc:     number of cells
+--- nt:     number of ARD products over time
+--- nodata: nodata value
++++ Return: void
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++**/
+void index_ratio_minus1(ard_t *ard, small *mask_, tsa_t *ts, int b1, int b2, int nc, int nt, short nodata){
+int p, t;
+float ind, scale = 10000.0;
+
+
+  #pragma omp parallel private(t,ind) shared(ard,mask_,ts,b1,b2,nc,nt,nodata,scale) default(none)
+  {
+
+    #pragma omp for
+    for (p=0; p<nc; p++){
+
+      if (mask_ != NULL && !mask_[p]){
+        for (t=0; t<nt; t++) ts->tss_[t][p] = nodata;
+        continue;
+      }
+
+      for (t=0; t<nt; t++){
+
+        if (!ard[t].msk[p]){
+          ts->tss_[t][p] = nodata;
+        } else {
+          ind = (ard[t].dat[b1][p]/ard[t].dat[b2][p])-1;
+          if (b2 == 0){
+            ts->tss_[t][p] = nodata;
+          } else {
+            ts->tss_[t][p] = (short)(ind*scale);
+          }
+        }
+
+      }
+
+    }
+  }
+
+  return;
+}
+
+
+/** This function computes an MSRre-like time series,
++++ MSRre: ((b1/b2)-1)/sqrt((b1/b2)+1)
+--- ard:    ARD
+--- mask_:  mask image
+--- ts:     pointer to instantly useable TSA image arrays
+--- b1:     band 1
+--- b2:     band 2
+--- nc:     number of cells
+--- nt:     number of ARD products over time
+--- nodata: nodata value
++++ Return: void
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++**/
+void index_msrre(ard_t *ard, small *mask_, tsa_t *ts, int b1, int b2, int nc, int nt, short nodata){
+int p, t;
+float upper, lower, ind, scale = 10000.0;
+
+
+  #pragma omp parallel private(t,upper,lower,ind) shared(ard,mask_,ts,b1,b2,nc,nt,nodata,scale) default(none)
+  {
+
+    #pragma omp for
+    for (p=0; p<nc; p++){
+
+      if (mask_ != NULL && !mask_[p]){
+        for (t=0; t<nt; t++) ts->tss_[t][p] = nodata;
+        continue;
+      }
+
+      for (t=0; t<nt; t++){
+
+        if (!ard[t].msk[p]){
+          ts->tss_[t][p] = nodata;
+        } else {
+          upper = (ard[t].dat[b1][p]/ard[t].dat[b2][p])-1;
+          lower = sqrt((ard[t].dat[b1][p]/ard[t].dat[b2][p])+1);
+          ind = upper/lower;
+          if (lower == 0){
```

Comment:
no idea about the valid range here...

probably add this to play it safe:
`|| ind*scale > SHRT_MAX || ind*scale < SHRT_MIN)){`

Reply:
Implemented in https://github.com/davidfrantz/force/pull/90/commits/62d9e4fed1b65ce65dad5bd2486975183deab8e9.